v1/topdown/cache/cache.go

```diff
@@ -394,16 +428,141 @@ func (c *interQueryValueCache) UpdateConfig(config *Config) {
 	c.config = config
 }
 
-func (c *interQueryValueCache) maxNumEntries() int {
+func (c *interQueryValueCacheBucket) maxNumEntries() int {
 	if c.config == nil {
 		return defaultInterQueryBuiltinValueCacheSize
 	}
-	return *c.config.InterQueryBuiltinValueCache.MaxNumEntries
+	return *c.config.MaxNumEntries
+}
+
+type InterQueryValueCache interface {
+	InterQueryValueCacheBucket
```

Comment:
Currently, the "root" cache bucket is a global cache shared between all built-ins. We could change this to be per-built-in specific, but hidden; i.e. the developer wouldn't need to know the name, but it would be separate from the cache returned to any other built-in.

Reply:
> Currently, the "root" cache bucket is a global cache shared between all built-ins.

So this is the regular value cache?

Reply:
> We could change this to be per-built-in specific, but hidden

Afair we don't advertise the implementation details anywhere so this could be fine. Any performance impact of the dedicated cache? For eg. if 2 builtins were to fully use the cache, it's 2x the memory, correct?

Reply:
> if 2 builtins were to fully use the cache, it's 2x the memory, correct
 
 Yes. Unless the caches are unbounded, then it wouldn't matter. 
If you have policies with e.g. a lot of regex and glob calculations and high throughput of concurrent queries, lock contention would likely be less with individual caches; but the difference is probably negligible.